src/main/java/com/commercetools/sync/services/UnresolvedReferencesService.java

```diff
@@ -14,30 +14,42 @@
      * Given product draft keys, this method fetches the persisted drafts waiting to be resolved. If there is no
      * matching draft, an empty {@link Set} will be returned in the returned future.
      *
-     * @param keys the keys of the persisted product drafts, waiting to be resolved, to fetch.
+     * @param keys the keys of the persisted  drafts, waiting to be resolved, to fetch.
+     * @param containerKey the key of the container, which contains the persisted draft
+     * @param clazz Class of the object contained by custom object
      * @return {@link CompletionStage}&lt;{@link Set}&gt; in which the result of its completion contains a
      *         {@link Set} that contains the matching drafts if any exist, otherwise empty.
      */
     @Nonnull
-    CompletionStage<Set<WaitingToBeResolved>> fetch(@Nonnull final Set<String> keys);
+    CompletionStage<Set<WaitingToBeResolved>> fetch(@Nonnull final Set<String> keys,
+                                                    @Nonnull final String containerKey,
+                                                    @Nonnull Class<? extends WaitingToBeResolved> clazz);
```

Comment:
Is it possible to use the type parameter instead of wildcard?

Reply:
no, I have to use a wildcard here